<!--
Author: Kurt Meister

Copyright (c) 2017, Kurt Meister
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

 * Redistributions of source code must retain the above copyright notice,
   this list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.

THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND ANY
EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED. IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE FOR ANY
DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
DAMAGE.

============================================================================

Description:
	launch file for the meister package. launches the obstacle detection
	system I developed for the ECE 573 project

-->

<!-- meister nodes -->
<node name="vehicle_status_monitor" pkg="meister" type="vehicle_status_monitor" output="screen">
    <param name="in_motion_threshold" value="0.75"/>
    <param name="stopped_threshold" value="0.1"/>
</node>



<node name="lidar_filter" pkg="meister" type="lidar_filter"/>
<node name="laser_data" pkg="meister" type="laser_data"/>

<node name="obstacle_detector" pkg="meister" type="obstacle_detector" output="screen">
<param name="world_template_filename" value="$(find meister)/worlds/empty.world"/>
<param name="world_output_filename" value="$(find meister)/worlds/detections.world"/>
</node>

<node name="state_update_node" pkg="meister" type="state_update_node" output="screen"/>
